src/gleam/erlang/file.gleam

```diff
@@ -111,32 +111,146 @@ pub type Reason {
   NotUtf8
 }
 
+/// TODO
+///
+pub type FileType {
+  Device
+  Directory
+  Other
+  Regular
+  Symlink
+}
+
+/// TODO
+///
+pub type Access {
+  NoAccess
+  Read
+  ReadWrite
+  Write
+}
+
+/// TODO
+///
+pub type FileInfo {
+  FileInfo(
+    size: Int,
+    file_type: FileType,
+    access: Access,
+    atime: Int,
+    mtime: Int,
+    ctime: Int,
+    mode: Int,
+    links: Int,
+    major_device: Int,
+    minor_device: Int,
+    inode: Int,
+    user_id: Int,
+    group_id: Int,
+  )
+}
+
+/// TODO
+///
+/// ## Examples
+///
+/// ```gleam
+/// ```
+///
+pub external fn read_file_info(path: String) -> Result(FileInfo, Reason) =
```

Comment:
Perhaps just `file_info`?

Reply:
info returning FileInfo sounds good to me

Reply:
I forgot about `link_info`, so I went with `file_info` to match. Also changed `exists` to `file_exists` and added `link_exists`.